File: app/src/main/java/com/nirbhor/app/ui/screens/more/InboxScreen.kt
package com.nirbhor.app.ui.screens

import androidx.compose.foundation.background
import androidx.compose.foundation.clickable
import androidx.compose.foundation.layout.Arrangement
import androidx.compose.foundation.layout.Box
import androidx.compose.foundation.layout.Column
import androidx.compose.foundation.layout.Row
import androidx.compose.foundation.layout.Spacer
import androidx.compose.foundation.layout.fillMaxSize
import androidx.compose.foundation.layout.fillMaxWidth
import androidx.compose.foundation.layout.height
import androidx.compose.foundation.layout.padding
import androidx.compose.foundation.layout.size
import androidx.compose.foundation.layout.width
import androidx.compose.foundation.rememberScrollState
import androidx.compose.foundation.shape.RoundedCornerShape
import androidx.compose.foundation.verticalScroll
import androidx.compose.material.icons.Icons
import androidx.compose.material.icons.filled.Inventory2
import androidx.compose.material.icons.filled.NotificationsActive
import androidx.compose.material.icons.filled.WarningAmber
import androidx.compose.material.icons.filled.EmojiEvents
import androidx.compose.material3.Icon
import androidx.compose.material3.Text
import androidx.compose.runtime.Composable
import androidx.compose.runtime.getValue
import androidx.compose.runtime.mutableStateOf
import androidx.compose.runtime.remember
import androidx.compose.runtime.setValue
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.draw.alpha
import androidx.compose.ui.draw.clip
import androidx.compose.ui.graphics.Color
import androidx.compose.ui.graphics.vector.ImageVector
import androidx.compose.ui.unit.dp
import androidx.lifecycle.compose.collectAsStateWithLifecycle
import com.nirbhor.app.navigation.LocalAppContainer
import com.nirbhor.app.navigation.NavActions
import com.nirbhor.app.ui.components.NirbhorTopBar
import com.nirbhor.app.ui.i18n.num
import com.nirbhor.app.ui.i18n.tr
import com.nirbhor.app.ui.theme.Dimens
import com.nirbhor.app.ui.theme.NirbhorTheme

private data class InboxItem(val icon: ImageVector, val title: String, val body: String, val amber: Boolean, val onClick: (() -> Unit)? = null)

/** Notification inbox (6b): day-grouped; unread carries a 4px warm left rule, read drops to 72%. */
@Composable
fun InboxScreen(actions: NavActions) {
    val colors = NirbhorTheme.colors
    val repo = LocalAppContainer.current.repository
    val stock by repo.stockStatuses().collectAsStateWithLifecycle(initialValue = emptyList())
    val meds by repo.medicines.collectAsStateWithLifecycle(initialValue = emptyList())
    var allRead by remember { mutableStateOf(false) }

    // Real, derived notifications only — one low-stock note per medicine that's running low.
    val lowMeds = meds.filter { m -> stock.firstOrNull { it.medicineId == m.id }?.isLow == true }
    val today = lowMeds.map { m ->
        InboxItem(
            Icons.Filled.Inventory2,
            tr("${m.displayName} ফুরিয়ে আসছে", "${m.displayName} running low"),
            tr("রিফিলে গিয়ে মজুত ঠিক করুন", "Open Refill to update stock"),
            amber = true, onClick = actions::openRefill,
        )
    }
    val earlier = emptyList<InboxItem>()

    Column(Modifier.fillMaxSize().background(colors.paper)) {
        NirbhorTopBar(
            title = tr("খবর", "Notifications"), onBack = actions::back,
            trailing = {
                Text(tr("সব পড়া হয়েছে", "Mark all read"), style = NirbhorTheme.type.cardTitleSecondary, color = colors.calm,
                    modifier = Modifier.clickable { allRead = true }.padding(6.dp))
            },
        )
        Column(
            Modifier.fillMaxSize().verticalScroll(rememberScrollState()).padding(horizontal = Dimens.screenPadding, vertical = 12.dp),
            verticalArrangement = Arrangement.spacedBy(18.dp),
        ) {
            if (today.isEmpty() && earlier.isEmpty()) {
                Spacer(Modifier.height(40.dp))
                Text(tr("এখন কোনো খবর নেই", "No notifications yet"), style = NirbhorTheme.type.cardTitlePrimary, color = colors.ink, modifier = Modifier.fillMaxWidth())
                Text(tr("ওষুধ ও মজুতের খবর এখানে আসবে।", "Dose and stock alerts will appear here."), style = NirbhorTheme.type.body, color = colors.ink3, modifier = Modifier.fillMaxWidth())
            } else {
                InboxGroup(tr("আজ", "Today"), today, read = allRead)
                InboxGroup(tr("আগে", "Earlier"), earlier, read = true)
            }
        }
    }
}

@Composable
private fun InboxGroup(label: String, items: List<InboxItem>, read: Boolean) {
    val colors = NirbhorTheme.colors
    if (items.isEmpty()) return
    Column(verticalArrangement = Arrangement.spacedBy(Dimens.cardGap)) {
        Text(label, style = NirbhorTheme.type.sectionLabel, color = colors.ink3)
        items.forEach { item ->
            Row(
                Modifier.fillMaxWidth().clip(RoundedCornerShape(Dimens.radiusCard)).background(colors.card)
                    .then(if (item.onClick != null) Modifier.clickable(onClick = item.onClick) else Modifier)
                    .alpha(if (read) 0.72f else 1f),
                verticalAlignment = Alignment.CenterVertically,
            ) {
                Box(Modifier.width(4.dp).height(56.dp).background(if (!read) colors.warm else Color.Transparent))
                Row(Modifier.padding(14.dp), verticalAlignment = Alignment.CenterVertically) {
                    Box(Modifier.size(38.dp).clip(RoundedCornerShape(11.dp)).background(if (item.amber) colors.warmSoft else colors.calmSoft), contentAlignment = Alignment.Center) {
                        Icon(item.icon, contentDescription = null, tint = if (item.amber) colors.warmD else colors.calmD, modifier = Modifier.size(20.dp))
                    }
                    Spacer(Modifier.width(12.dp))
                    Column {
                        Text(item.title, style = NirbhorTheme.type.cardTitleSecondary, color = colors.ink)
                        Text(item.body, style = NirbhorTheme.type.meta, color = colors.ink3)
                    }
                }
            }
        }
    }
}

File: lib/ui/screens/inbox_screen.dart
import 'package:flutter/material.dart';

import '../../data/app_scope.dart';
import '../../domain/models.dart';
import '../../navigation/nav_actions.dart';
import '../../theme/theme.dart';
import '../components/scaffold.dart';

/// Stable signature of the notifications the inbox derives, so "mark all read" and the Home bell's
/// unread dot are computed from exactly one definition and cannot drift apart.
String inboxSignature(List<Medicine> medicines, List<StockStatus> stock) {
  final low = stock.where((s) => s.isLow).map((s) => s.medicineId).toSet();
  final ids = medicines.where((m) => low.contains(m.id)).map((m) => m.id).toList()..sort();
  return ids.join('|');
}

/// Whether the inbox holds anything the patient has not yet marked read.
bool hasUnreadInbox(List<Medicine> medicines, List<StockStatus> stock, String readSignature) {
  final signature = inboxSignature(medicines, stock);
  return signature.isNotEmpty && signature != readSignature;
}

class _InboxItem {
  const _InboxItem({
    required this.icon,
    required this.title,
    required this.body,
    required this.amber,
    this.onTap,
  });

  final IconData icon;
  final String title;
  final String body;
  final bool amber;
  final VoidCallback? onTap;
}

/// Notification inbox (6b): day-grouped; unread carries a 4px warm left rule, read drops to 72%.
class InboxScreen extends StatelessWidget {
  const InboxScreen({super.key});

  @override
  Widget build(BuildContext context) {
    final colors = context.colors;
    final settingsStore = context.settingsStore;

    return Scaffold(
      backgroundColor: colors.paper,
      body: ListenableBuilder(
        listenable: settingsStore,
        builder: (context, _) => RepoBuilder<(List<Medicine>, List<StockStatus>)>(
          query: (repo) async => (await repo.medicines(), await repo.stockStatuses()),
          builder: (context, data) {
            final (meds, stock) = data;
            // Real, derived notifications only — one low-stock note per medicine running low.
            final lowMeds = meds
                .where((m) => stock.any((s) => s.medicineId == m.id && s.isLow))
                .toList();
            final today = lowMeds
                .map((m) => _InboxItem(
                      icon: Icons.inventory_2,
                      title: context.tr(
                        '${m.displayName} ফুরিয়ে আসছে',
                        '${m.displayName} running low', hi: '${m.displayName} कम पड़ रही है', es: 'Queda poco de ${m.displayName}',
                      ),
                      body: context.tr('রিফিলে গিয়ে মজুত ঠিক করুন', 'Open Refill to update stock'),
                      amber: true,
                      onTap: context.nav.openRefill,
                    ))
                .toList();
            const earlier = <_InboxItem>[];

            final signature = inboxSignature(meds, stock);
            final allRead =
                signature.isNotEmpty && settingsStore.value.inboxReadSignature == signature;
            final canMarkRead = signature.isNotEmpty && !allRead;

            return Column(
              children: [
                NirbhorTopBar(
                  title: context.tr('খবর', 'Notifications'),
                  onBack: context.nav.back,
                  trailing: InkWell(
                    borderRadius: BorderRadius.circular(10),
                    onTap: canMarkRead
                        ? () => settingsStore.setInboxReadSignature(signature)
                        : null,
                    child: Container(
                      constraints: const BoxConstraints(minHeight: Dimens.tapMin),
                      alignment: Alignment.center,
                      padding: const EdgeInsets.symmetric(horizontal: 8, vertical: 14),
                      child: Text(
                        context.tr('সব পড়া হয়েছে', 'Mark all read'),
                        style: context.type.cardTitleSecondary.copyWith(
                          // Dim when there is nothing left to mark, so the tap target reads as inert.
                          color: canMarkRead ? colors.calm : colors.ink3,
                        ),
                      ),
                    ),
                  ),
                ),
                Expanded(
                  child: SingleChildScrollView(
                    padding: const EdgeInsets.symmetric(
                      horizontal: Dimens.screenPadding,
                      vertical: 12,
                    ),
                    child: SafeArea(
                      top: false,
                      child: today.isEmpty && earlier.isEmpty
                          ? Column(
                              crossAxisAlignment: CrossAxisAlignment.stretch,
                              children: [
                                const SizedBox(height: 40),
                                Text(
                                  context.tr('এখন কোনো খবর নেই', 'No notifications yet'),
                                  style: context.type.cardTitlePrimary.copyWith(color: colors.ink),
                                ),
                                const SizedBox(height: 6),
                                Text(
                                  context.tr(
                                    'ওষুধ ও মজুতের খবর এখানে আসবে।',
                                    'Dose and stock alerts will appear here.',
                                  ),
                                  style: context.type.body.copyWith(color: colors.ink3),
                                ),
                              ],
                            )
                          : Column(
                              crossAxisAlignment: CrossAxisAlignment.stretch,
                              children: [
                                _InboxGroup(
                                  label: context.tr('আজ', 'Today'),
                                  items: today,
                                  read: allRead,
                                ),
                                const _InboxGroup(label: '', items: earlier, read: true),
                              ],
                            ),
                    ),
                  ),
                ),
              ],
            );
          },
        ),
      ),
    );
  }
}

class _InboxGroup extends StatelessWidget {
  const _InboxGroup({required this.label, required this.items, required this.read});

  final String label;
  final List<_InboxItem> items;
  final bool read;

  @override
  Widget build(BuildContext context) {
    if (items.isEmpty) return const SizedBox.shrink();
    final colors = context.colors;
    return Padding(
      padding: const EdgeInsets.only(bottom: 18),
      child: Column(
        crossAxisAlignment: CrossAxisAlignment.stretch,
        children: [
          Text(label, style: context.type.sectionLabel.copyWith(color: colors.ink3)),
          const SizedBox(height: Dimens.cardGap),
          for (final item in items)
            Padding(
              padding: const EdgeInsets.only(bottom: Dimens.cardGap),
              child: Opacity(
                opacity: read ? 0.72 : 1,
                child: Material(
                  color: colors.card,
                  borderRadius: BorderRadius.circular(Dimens.radiusCard),
                  clipBehavior: Clip.antiAlias,
                  child: InkWell(
                    onTap: item.onTap,
                    child: IntrinsicHeight(
                      child: Row(
                        children: [
                          Container(
                            width: 4,
                            color: read ? Colors.transparent : colors.warm,
                          ),
                          Expanded(
                            child: Padding(
                              padding: const EdgeInsets.all(14),
                              child: Row(
                                children: [
                                  Container(
                                    width: 38,
                                    height: 38,
                                    decoration: BoxDecoration(
                                      color: item.amber ? colors.warmSoft : colors.calmSoft,
                                      borderRadius: BorderRadius.circular(11),
                                    ),
                                    child: Icon(
                                      item.icon,
                                      size: 20,
                                      color: item.amber ? colors.warmD : colors.calmD,
                                    ),
                                  ),
                                  const SizedBox(width: 12),
                                  Expanded(
                                    child: Column(
                                      crossAxisAlignment: CrossAxisAlignment.start,
                                      mainAxisSize: MainAxisSize.min,
                                      children: [
                                        Text(
                                          item.title,
                                          style: context.type.cardTitleSecondary
                                              .copyWith(color: colors.ink),
                                        ),
                                        Text(
                                          item.body,
                                          style: context.type.meta.copyWith(color: colors.ink3),
                                        ),
                                      ],
                                    ),
                                  ),
                                ],
                              ),
                            ),
                          ),
                        ],
                      ),
                    ),
                  ),
                ),
              ),
            ),
        ],
      ),
    );
  }
}
